src/Symfony/Component/HttpFoundation/File/UploadedFile.php

```diff
@@ -236,6 +236,24 @@ public function move($directory, $name = null)
         throw new FileException($this->getErrorMessage());
     }
 
+    public function moveAndRename(string $targetDirectory, string $fileNamePattern)
+    {
+        $fileName = strtr($fileNamePattern, [
+            '[contenthash]' => sha1_file($this->getRealPath()),,
+            '[day]' => date('d'),
+            '[extension]' => $this->getExtension(),
+            '[month]' => date('m'),
+            '[name]' => $this->getBasename(),
+            '[randomhash]' => bin2hex(random_bytes(20)),
+            '[slug]' => transliterator_transliterate('Any-Latin; Latin-ASCII; [^A-Za-z0-9_] remove; Lower()', $this->getBasename()),
```

Comment:
this requires the intl-extension, which is not stubbed by symfony/intl (nor is a composer dev-dependency for this component)

Reply:
and the way things are implemented, this is a dependency even if they don't use the `[slug]` placeholder.